### PR TITLE
Qt5PatchCollection snapshots

### PR DESCRIPTION
**Pull request by a17r**
**tl;dr:** bumps based on current state of Qt overlay, following discussion in https://github.com/gentoo/qt/pull/238, with EAPI-8 and minor changes listed below.

qt5-build.eclass: Support EAPI-8, drop eapi8-dosym

dev-qt/qtcore:
- Retain fallback RDEPEND on dev-qt/qtchooser in place of other dev-qt/
- Raise minimum version to >=dev-qt/qtchooser-66-r1 for smooth upgrade
- Drop revert to unsupported timezone handling (bug 737914)
  See also: https://archives.gentoo.org/gentoo-dev/message/417bd48a17042665cfdf2e0a1775ca52
  TODO: pkg_postinst ewarn?

dev-qt/qtchooser: Move config file install from qt5-build.eclass
- Block colliding <dev-qt/qtcore-5.15.2-r10

dev-qt/qtdeclarative:
- python3_10

PS: This currently resides in kde overlay as long as kde.org.eclass in Gentoo ebuild repo does not have EAPI-8 support.
PPS: This omits packages w/o relevant changes, including the complete qttools module.

### CODE REVIEW

**Comment by a17r** [comment]
Moved to https://github.com/gentoo/gentoo/pull/22039.
